pandapower/io_utils.py

```diff
@@ -746,6 +748,23 @@ def from_json(cls, json_string):
         d = json.loads(json_string, cls=PPJSONDecoder)
         return cls.from_dict(d)
 
+    def __eq__(self, other):
+        """
+        isinstance: so that we know we can use to_json()
+        comparing class name instead of class directly allows more flexibility,
+        e.g. when the class definition is moved to a different module.
+        There is still a risk that the implementation details of the methods can differ
+        if the classes are from different modules.
+        Comparison is based on comparing dictionaries of the classes.
+        To this end, the dictionary comparison library deepdiff is used for recursive comparison.
+        """
+        if not isinstance(other, self.__class__) or self.__class__.__name__ != other.__class__.__name__:
+            return False
+        else:
+            d = DeepDiff(self, other, ignore_order=True, ignore_nan_inequality=True, significant_digits=6,
```

Comment:
No clue what is happening here, but I trust you.

Reply:
In the first condition:
* isinstance checks if the "other" object is the same class or is a subclass of that class (e.g. isinstance(ContinuousTapControl, TrafoController) will be True
* if it is, then we can also compare the class names (at this point we know that the .__class__ attribute will exist because otherwise isinstance(...) would already stop the if ... check)
* this part makes an early return if either of the conditions is False

The second part uses DeepDiff to recursively compare the dictionaries of the objects. I don't think it compares the methods of the objects, just attributes.

```python
c = pp.io_utils.JSONSerializableClass()
c2 = pp.io_utils.JSONSerializableClass()
c2.hello = "hi"
DeepDiff(c, c2)
Out[15]: {'attribute_added': [root.hello]}
```

Reply:
It seems like it also compares the classes. But we cannot see the difference in terms of the added method "hello":

```python
class MyTest(pp.io_utils.JSONSerializableClass):
    def hello(self):
        return "hi"
    
c3 = MyTest()
DeepDiff(c, c3)
Out[18]: 
{'type_changes': {'root': {'old_type': pandapower.io_utils.JSONSerializableClass,
   'new_type': __main__.MyTest,
   'old_value': <pandapower.io_utils.JSONSerializableClass at 0x266109dc0d0>,
   'new_value': <__main__.MyTest at 0x266109457b0>}}}
```

Reply:
The parameter "significant_digits" specifies the number of digits after the decimal point to check for. I set it to 6. One could also provide a tolerance "math_epsilon" for "isclose". I think I will add it and also set to 1e-6.